fontique/src/generic.rs

```diff
@@ -148,3 +173,78 @@ impl GenericFamilyMap {
         self.map[generic as usize].extend(families);
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use crate::GenericFamily;
+    use bytemuck::{checked::try_from_bytes, Contiguous, Zeroable};
+    use core::ptr;
+
+    #[test]
+    fn checked_bit_pattern() {
+        let valid = bytemuck::bytes_of(&2_u8);
+        let invalid = bytemuck::bytes_of(&200_u8);
+
+        assert_eq!(
+            Ok(&GenericFamily::Monospace),
+            try_from_bytes::<GenericFamily>(valid)
+        );
+
+        assert!(try_from_bytes::<GenericFamily>(invalid).is_err());
+    }
+
+    #[test]
+    fn contiguous() {
+        let hd1 = GenericFamily::SansSerif;
+        let hd2 = GenericFamily::from_integer(hd1.into_integer());
+        assert_eq!(Some(hd1), hd2);
+
+        assert_eq!(None, GenericFamily::from_integer(255));
+    }
+
+    #[test]
+    fn zeroable() {
+        let hd = GenericFamily::zeroed();
+        assert_eq!(hd, GenericFamily::Serif);
+    }
+
+    /// Tests that the [`Contiguous`] impl for [`GenericFamily`] is not trivially incorrect.
+    const _: () = {
+        let mut value = 0;
+        while value <= GenericFamily::MAX_VALUE {
+            // Safety: In a const context, therefore if this makes an invalid GenericFamily, that will be detected.
+            // When updating the MSRV to 1.82 or later, this can use `&raw const value` instead of the addr_of!
+            let it: GenericFamily = unsafe { ptr::read((core::ptr::addr_of!(value)).cast()) };
+            // Evaluate the enum value to ensure it actually has a valid tag
+            if it as u8 != value {
+                unreachable!();
+            }
+            value += 1;
+        }
+    };
+}
+
+#[cfg(doctest)]
+/// Doctests aren't collected under `cfg(test)`; we can use `cfg(doctest)` instead
+mod doctests {
+    /// Validates that any new variants in `GenericFamily` has led to a change in the `Contiguous` impl.
+    /// Note that to test this robustly, we'd need 256 tests, which is impractical.
+    /// We make the assumption that all new variants will maintain contiguousness.
+    ///
+    /// ```compile_fail,E0080
+    /// use bytemuck::Contiguous;
+    /// use styled_text::GenericFamily;
```

Comment:
Hmm. I don't think that this is the right import path...

Indeed, this test fails on nightly for me.

Reply:
Whoops, copy/paste from my branch. Fixed and force pushed.